### ISSUE TITLE
Test failures due to RuleContext missing new token methods

### ISSUE THREAD

**Issue by btmills** [eslint/eslint#750]
This gives me a weird feeling since nobody else has reported it, and Travis has the build passing, but I tested on my dev machine and on a fresh checkout from GitHub on a brand new VM, and I'm getting identical errors on both. 260 tests are failing, all with error messages related to #738:

```
TypeError: Object #<RuleContext> has no method 'getTokenBefore'
```

Same with `getFirstTokens`, `getLastToken`, and `getTokenAfter` as I scrolled through part of the list.

I looked into the properties of the `context` object and confirmed that sometimes it does indeed appear to be missing those members:

``` js
// Normal
{ getSource: [Function],
  getTokens: [Function],
  getTokensBefore: [Function],
  getTokenBefore: [Function],
  getTokensAfter: [Function],
  getTokenAfter: [Function],
  getFirstTokens: [Function],
  getFirstToken: [Function],
  getLastTokens: [Function],
  getLastToken: [Function],
  getComments: [Function],
  getAncestors: [Function],
  getScope: [Function],
  getJSDocComment: [Function],
  getFilename: [Function],
  report: [Function] }

// Failing
{ getSource: [Function],
  getTokens: [Function],
  getComments: [Function],
  getAncestors: [Function],
  getScope: [Function],
  getJSDocComment: [Function],
  getFilename: [Function],
  report: [Function] }
```

Everything is fine when I checkout 4ab802b, but as of 40ed626 I'm seeing this behavior. I'm going to keep digging, but is anyone else seeing this?
